src/cluster.c

```diff
@@ -4578,12 +4661,25 @@ sds clusterGenNodeDescription(clusterNode *node, int use_pport) {
 
     /* Node coordinates */
     ci = sdscatlen(sdsempty(),node->name,CLUSTER_NAMELEN);
-    if (node->hostname) {
+    if (node->hostname && node->nodename) {
```

Comment:
If we're going to do this, we should probably just call sdscatfmt multiple times, incrementally adding on the data.